opentelemetry_configuration.json

```diff
@@ -520,6 +520,113 @@
         }
       }
     },
+    "ExperimentalComposableRuleBasedSampler": {
+      "type": [
+        "object",
+        "null"
+      ],
+      "additionalProperties": false,
+      "properties": {
+        "rules": {
+          "type": [
+            "array",
+            "null"
+          ],
+          "items": {
+            "$ref": "#/$defs/ExperimentalComposableRuleBasedSamplerRule"
+          },
+          "description": "The rules for the sampler, matched in order. Each rule can have multiple match conditions - the sampler will be applied if all match. \nIf no conditions are specified, the rule matches all spans that reach it. If no rules match, the span is not sampled.\n"
+        }
+      }
+    },
+    "ExperimentalComposableRuleBasedSamplerRule": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "attribute_values": {
+          "$ref": "#/$defs/ExperimentalComposableRuleBasedSamplerRuleAttributeValues",
+          "description": "Values to match against a single attribute. Non-string attributes are matched using their string representation:\nfor example, a value of \"404\" would match the http.response.status_code 404. For array attributes, if any\nitem matches, it is considered a match.\n"
+        },
+        "attribute_patterns": {
+          "$ref": "#/$defs/ExperimentalComposableRuleBasedSamplerRuleAttributePatterns",
+          "description": "Patterns to match against a single attribute. Non-string attributes are matched using their string representation:\nfor example, a pattern of \"4*\" would match any http.response.status_code in 400-499. For array attributes, if any\nitem matches, it is considered a match.\n"
+        },
+        "span_kinds": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "$ref": "#/$defs/SpanKind"
+          },
+          "description": "The span kinds to match. If the span's kind matches any of these, it matches."
+        },
+        "parent": {
```

Comment:
I found that a useful predicate for rule based sampling is `no_local_parent`, which nicely identifies service head spans. This, of course, in addition to `is_trace_root`.  I realize that the proposed `parent` rule is expressive enough, but it is more verbose.

Reply:
I went with keeping the parent enum since then we don't have to worry about "what if both are set" type of situations. I guess in terms of verbosity, `parent: LOCAL` is similar to `no_local_parent: true`? Possibly helped by removing the SAMPLED state from the enum, let me know if it looks fine now